src/libraries/System.Text.Json/src/System/Text/Json/Serialization/WriteStackFrame.cs

```diff
@@ -18,6 +18,11 @@ internal struct WriteStackFrame
         /// </summary>
         public IEnumerator? CollectionEnumerator;
 
+        /// <summary>
+        /// The enumerator for resumable disposables.
+        /// </summary>
+        public IDisposable? Disposable;
```

Comment:
If this is already stored in `CollectionEnumerator`, why do we need a separate field?

Reply:
I agree, we do not need it. I followed a little bit too closely the async path.